mcstatus/status_response.py

```diff
@@ -0,0 +1,640 @@
+            return self.list
+
+    class Version(JavaStatusVersion):
+        """Deprecated class for `version` field.
+
+        Use `JavaStatusVersion` instead.
+        """
+
+        @overload
+        def __init__(self, raw: Dict[str, Any]) -> None:
+            ...
+
+        @overload
+        def __init__(self, name: str, protocol: int) -> None:
+            ...
+
+        def __init__(self, *args, **kwargs) -> None:
+            try:
+                bound = Signature(
+                    parameters=[
+                        Parameter("raw", Parameter.POSITIONAL_OR_KEYWORD, annotation=Dict[str, Any]),
+                    ]
+                ).bind(*args, **kwargs)
+
+                deprecated(
+                    lambda: None,
+                    replacement="mcstatus.status_response.JavaStatusVersion",
+                    date="2022-08",
+                    display_name="JavaStatusResponse.Version",
+                )()
+
+                # build and copy instance from `built` method
+                super().__init__(**self.build(bound.arguments["raw"]).__dict__)
+            except TypeError:
+                super().__init__(*args, **kwargs)
+
+    players: Players
+    version: Version
+
+    @overload
+    def __init__(self, raw: Dict[str, Any]) -> None:
+        ...
+
+    @overload
+    def __init__(
+        self,
+        players: JavaStatusPlayers,
+        version: JavaStatusVersion,
+        motd: str,
+        latency: float,
+        icon: Optional[str],
+    ) -> None:
+        ...
+
+    def __init__(self, *args, **kwargs) -> None:
+        try:
+            bound = Signature(
+                parameters=[
+                    Parameter("raw", Parameter.POSITIONAL_OR_KEYWORD, annotation=Dict[str, Any]),
+                ]
+            ).bind(*args, **kwargs)
+
+            deprecated(lambda: None, replacement="build", date="2022-08", display_name="JavaStatusResponse.__init__")()
+
+            instance = self.build(bound.arguments["raw"]).__dict__
+            instance["players"] = self.Players(
+                online=instance["players"].online,
+                max=instance["players"].max,
+                list=instance["players"].list,
+            )
+            instance["version"] = self.Version(
+                name=instance["version"].name,
+                protocol=instance["version"].protocol,
+            )
+            super().__init__(**instance)
+        except TypeError:
+            super().__init__(*args, **kwargs)
+            # re-def fields to support the old interface
+            self.players = self.Players(
+                online=self.players.online,
+                max=self.players.max,
+                list=self.players.list,
+            )
+            self.version = self.Version(
+                name=self.version.name,
+                protocol=self.version.protocol,
+            )
+
+    @property
+    @deprecated(replacement="motd", date="2022-08")
+    def description(self):
```

Comment:
Provide return types for all of these properties, they're what gives type-checekrs the information about what these property parameters will be returning
```suggestion
    def description(self) -> str:
```

Reply:
You should configure `pyright` to report not-typed functions.

Reply:
Fixed in aa71069b217593c10d5de91e6e12cafb8461e3c6.

Reply:
We could add `flake8-annotations` which is probably a better solution, because we would want to exempt tests from this, but not from all pyright checks. I can do this in another PR